mp4parse/src/lib.rs

```diff
@@ -893,6 +1139,17 @@ fn read_fullbox_extra<T: ReadBytesExt>(src: &mut T) -> Result<(u8, u32)> {
         u32::from(flags_a) << 16 | u32::from(flags_b) << 8 | u32::from(flags_c)))
 }
 
+// Parse the extra fields for a full box whose flag fields must be zero.
+fn read_fullbox_version_no_flags<T: ReadBytesExt>(src: &mut T) -> Result<u8> {
```

Comment:
The idea here was that there were several cases (4 just in the code that I added) where it wasn't that the caller didn't _care_ about flags, but that the spec required they be 0. So rather than repeat
```rust
    if flags != 0 {
        return Err(Error::Unsupported("expected flags to be 0"));
    }
```
4 times, or not enforce the flag requirement, it seemed worth packaging that in a function.

If you think it would be better to not enforce that requirement and truly ignore the flags, I'm happy to make that change; just let me know.